AudioFile.cpp

```diff
@@ -603,6 +609,21 @@ bool AudioFile<T>::saveToWaveFile (std::string filePath)
                 fileData.push_back (bytes[1]);
                 fileData.push_back (bytes[2]);
             }
+            else if (bitDepth == 32)
+            {
+              int32_t sampleAsIntAgain = (int32_t) (samples[channel][i] * (T)(1<<32));
```

Comment:
Need to look in more detail when it isn't 1am but potentially an issue here with the bit shift - i.e. `1 << 32` is larger than a `int32_t` right? So this should be something that can be multiplied by 1 or -1 and still fit into the int32_t. So if a int32_t range is -2147483648 to 2147483647, maybe it should be multiply by 2147483647? This would be `INT_MAX` again as your above code uses, but probably best to hard code it here in case of some unforeseen future system that decides to have a different value for `INT_MAX`?

Reply:
Absolutely valid point you make -- but this was the same kind of typo as above -- it's supposed to be `1 << 31` -- corrected now.

Reply:
I think we need to be using the max value for an int32 here again. `1 << 31` is `-2147483648` which isn't the value we need here. So it should be...

`int32_t sampleAsIntAgain = (int32_t) (samples[channel][i] * (T)std::numeric_limits<std::int32_t>::max());`

Sound right? :)

Reply:
(This thread is the `1 << 31` thing ^^^)

Reply:
Oh yes, you're absolutely right -- it should be `(1 << 31) - 1 = 2147483647`, which is `std::numeric_limits<std::int32_t>::max()`

I made the changes -- and double checked in the debugger that `+1.0` is mapped to `2147483647` and `-1.0` to `-2147483647`.

Thanks for your review!